### PR TITLE
Add facter 3 

### PR DESCRIPTION
**Pull request by DavidS**

### CODE REVIEW

**Comment by mcanevet** [comment]
@DavidS thanks a lot.
Shouldn't we use `--show-legacy` flag available since facter 3.0.2 to also show unstructured facts?
I was wondering why PL does not provider facter 3 as a gem, it should be possible even it's now coded in C++, doesn't it?

**Comment by DavidS** [comment]
`--show-legacy` is a good idea, but long-term modules and infra projects like here need to support both

re facter3 as gem: possible, yes. probable, no.

**Comment by mcanevet** [comment]
@DavidS I think that all tests towards facter3 will fail for now if we don't add `--show-legacy`, so it should be good to add it.
I really don't understand we it is not probable that PL release facter3 as a gem, it is really not convenient to install a package to test a specific version of facter...

File: facts/get_facts.sh [hunk 1]
```diff
@@ -12,6 +14,7 @@ bundle install --path vendor/bundler
 operatingsystem=$(bundle exec facter operatingsystem | tr '[:upper:]' '[:lower:]')
 operatingsystemrelease=$(bundle exec facter operatingsystemrelease)
 operatingsystemmajrelease=$(bundle exec facter operatingsystemmajrelease)
+lsbdistcodename=$(bundle exec facter lsbdistcodename)
```
Comment (mcanevet) [inline facts/get_facts.sh]: Unless I'm wrong, `lsb*` facts are not available by default on RedHat.

File: facts/get_facts.sh [hunk 2]
```diff
@@ -40,3 +43,42 @@ for version in 1.6.0 1.7.0 2.0.0 2.1.0 2.2.0 2.3.0 2.4.0; do
...
+
+case "${operatingsystem}" in
+  redhat)
+    case "${operatingsystemmajrelease}" in
+      5)
+        curl "https://yum.puppetlabs.com/puppetlabs-release-pc1-el-${operatingsystemmajrelease}.noarch.rpm" > /tmp/puppetlabs-release-pc1.rpm
+        rpm -ivh /tmp/puppetlabs-release-pc1.rpm ;;
+      *)
+        rpm -ivh "https://yum.puppetlabs.com/puppetlabs-release-pc1-el-${operatingsystemmajrelease}.noarch.rpm"
+        yum -y install puppet-agent
```
Comment (mcanevet) [inline facts/get_facts.sh]: You don't install puppet-agent on RedHat5?

File: facts/get_facts.sh [hunk 3]
```diff
@@ -40,3 +43,42 @@ for version in 1.6.0 1.7.0 2.0.0 2.1.0 2.2.0 2.3.0 2.4.0; do
...
+        ;;
+    esac
+
+    /opt/puppetlabs/bin/facter -j | tee $output_file
```
Comment (mcanevet) [inline facts/get_facts.sh]: Could be moved out of the  ̀case`

Comment (DavidS) [inline facts/get_facts.sh]: goot point. I didn't invest enough time to actually run it on rhel.

**Comment by mcanevet** [comment]
@DavidS I merge this even if it may need improvements.
Thanks

**Pull request merged**